packages/atomic/src/components/commerce/atomic-commerce-products-per-page/atomic-commerce-products-per-page.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  Pagination,
+  PaginationState,
+  buildSearch,
+  buildProductListing,
+  SearchSummaryState,
+  ProductListingSummaryState,
+  Summary,
+  loadPaginationActions,
+} from '@coveo/headless/commerce';
+import {Component, Event, EventEmitter, h, Prop, State} from '@stencil/core';
+import {
+  BindStateToController,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../../utils/initialization-utils';
+import {randomID} from '../../../utils/utils';
+import {FieldsetGroup} from '../../common/fieldset-group';
+import {Choices} from '../../common/items-per-page/choices';
+import {Label} from '../../common/items-per-page/label';
+import {
+  validateChoicesDisplayed,
+  validateInitialChoice,
+} from '../../common/items-per-page/validate';
+import {PagerGuard} from '../../common/pager/pager-guard';
+import type {CommerceBindings as Bindings} from '../atomic-commerce-interface/atomic-commerce-interface';
+
+/**
+ * The `atomic-commerce-products-per-page` component determines how many products to display per page.
+ *
+ * @part label - The "Products per page" label.
+ * @part buttons - The list of buttons.
+ * @part button - The product per page button.
+ * @part active-button - The active product per page button.
+ *
+ * @alpha
+ */
+@Component({
+  tag: 'atomic-commerce-products-per-page',
+  styleUrl: 'atomic-commerce-products-per-page.pcss',
+  shadow: true,
+})
+export class AtomicCommerceProductsPerPage
+  implements InitializableComponent<Bindings>
+{
+  @InitializeBindings() public bindings!: Bindings;
+
+  public pagination!: Pagination;
+  @BindStateToController('pagination')
+  @State()
+  private paginationState!: PaginationState;
+
+  public summary!: Summary;
+  @BindStateToController('summary')
+  @State()
+  private summaryState!: SearchSummaryState | ProductListingSummaryState;
+
+  @State() public error!: Error;
+  private choices!: number[];
+  private readonly radioGroupName = randomID(
+    'atomic-commerce-products-per-page-'
+  );
+
+  /**
+   * A list of choices for the number of products to display per page, separated by commas.
+   */
+  @Prop({reflect: true}) choicesDisplayed = '10,25,50,100';
+  /**
+   * The initial selection for the number of product per page. This should be part of the `choicesDisplayed` option. By default, this is set to the first value in `choicesDisplayed`.
+   */
+  @Prop({mutable: true, reflect: true}) initialChoice?: number;
```

Comment:
I messed around a bit to try to understand why your e2e tests are failing, and I found out that as it is, in storybook, the `initialChoice` property is typed as an object (which makes no sense!)

 if you set a default value (say 0) to the initialChoice, then everything works fine (you don't even have to adjust the check in validate.ts, cause !context.initialChoice will be truthy when initialChoice === 0).

Reply:
Yea it is weird. I found a fix where adding a @type {number} in the jsdoc fixes it !